packages/next/src/server/lib/start-server.ts

```diff
@@ -38,6 +39,73 @@ import type { ConfiguredExperimentalFeature } from '../config'
 const debug = setupDebug('next:start-server')
 let startServerSpan: Span | undefined
 
+/**
+ * Get the process ID (PID) of the process using the specified port
+ */
+async function getProcessIdUsingPort(port: number): Promise<string | null> {
+  const timeoutMs = 250
+  const processLookupController = new AbortController()
+
+  const pidPromise = new Promise<string | null>((resolve) => {
+    const handleError = (error: Error) => {
+      debug('Failed to get process ID for port', port, error)
+      resolve(null)
+    }
+
+    try {
+      // Use lsof on Unix-like systems (macOS, Linux)
+      if (process.platform !== 'win32') {
+        exec(
+          `lsof -ti:${port}`,
```

Comment:
This also includes processes that have established connections which is undesired. Following up in https://github.com/vercel/next.js/pull/81903